File: src/main/java/com/howell/matt/test/LambdaExamples.java
package com.howell.matt.test;

import com.amazonaws.regions.Regions;
import com.amazonaws.services.lambda.AWSLambda;
import com.amazonaws.services.lambda.AWSLambdaClientBuilder;
import com.amazonaws.services.lambda.model.FunctionConfiguration;

import lombok.extern.slf4j.Slf4j;

/**
 * Lambda examples.
 */
@Slf4j
public class LambdaExamples {

    /**
     * Main.
     *
     * @param args args
     */
    public static void main(String[] args) {
        new LambdaExamples().showLambdaFunctions();
    }

    /**
     * Show lambda functions.
     */
    public void showLambdaFunctions() {
        final AWSLambda lambdaClient = AWSLambdaClientBuilder.standard().withRegion(Regions.US_WEST_2).build();
        lambdaClient.listFunctions().getFunctions().stream().map(FunctionConfiguration::getFunctionName).forEach
            (log::info);
    }
}

File: src/main/kotlin/com/howell/matt/kotlin/test/LambdaExamples.kt
package com.howell.matt.kotlin.test

import com.amazonaws.regions.Regions
import com.amazonaws.services.lambda.AWSLambdaClientBuilder

/**
 * Lambda examples.
 */
class LambdaExamples {

    val log = loggerFor(::LambdaExamples.javaClass)

    /**
     * Show lambda functions.
     */
    fun showLambdaFunctions() {
        val lambdaClient = AWSLambdaClientBuilder.standard().withRegion(Regions.US_WEST_2).build()
        lambdaClient.listFunctions().functions.stream().map({ it.getFunctionName() }).forEach({ log.info(it) })
    }

    companion object {
        /**
         * Main.
         *
         * @param args args
         */
        @JvmStatic fun main(args: Array<String>) {
            LambdaExamples().showLambdaFunctions()
        }
    }
}
